homeassistant/components/obihai/sensor.py

```diff
@@ -146,19 +187,25 @@ def icon(self):
             return "mdi:restart-alert"
         return "mdi:phone"
 
-    def update(self) -> None:
+    def update(self) -> bool:
         """Update the sensor."""
-        services = self._pyobihai.get_state()
+        if self._pyobihai.check_account():
```

Comment:
You can return early here instead, and un-indent:
```suggestion
        if not self._pyobihai.check_account():
            self._state = None
            return
```